### PR TITLE
#473 added date on hover for project changes

### PR DESCRIPTION
**Pull request by meisbobzheng**
## Changes

- Added a tooltip on hover for the project changes list. It shows the date implemented as well as the user that implemented it. 
- Revised the backend for an implementedChange to include the dateImplemented
- Revised test cases to fit for the changes to the backend

## Notes

First ticket, so I'm not really sure what I'm doing but it works. Reid helped a lot. 

## Test Cases

- Used altered existing test data with dateImplemented
- Manually checked the site locally and everything was working

## Screenshots (if applicable)

image

## To Do

- Should be all done for this ticket, but it is part of a group of tickets for visual enhancements

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://github.com/Northeastern-Electric-Racing/PM-Dashboard-v2/blob/main/docs/ContributorGuide.md) and reach out to your squad if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [X] All commits are tagged with the ticket number
- [X] No linting errors
- [X] No newline at end of file warnings
- [X] All code follows repository-configured prettier formatting
- [X] No merge conflicts
- [X] All checks passing
- [X] Screenshots of UI changes (if applicable)
- [X] Remove any not-applicable sections
- [X] Assign the PR to yourself
- [X] PR is linked to the ticket
- [X] No `package-lock.json` changes (unless dependencies have changed)
- [X] Request reviewers & ping on Slack

Closes # (#473 )

### CODE REVIEW

File: src/components/projects/wbs-details/work-package-container/changes-list/changes-list.tsx [hunk 1]
```diff
@@ -20,8 +22,17 @@ const ChangesList: React.FC<ChangesListProps> = ({ changes }) => {
       headerRight={<></>}
       list={changes.map((ic) => (
         <>
-          [<Link to={`${routes.CHANGE_REQUESTS}/${ic.changeRequestId}`}>#{ic.changeRequestId}</Link>
-          ] {ic.detail}
+          [<Link to={`${routes.PROJECTS}/${wbsPipe(ic.wbsNum)}`}>{wbsPipe(ic.wbsNum)}</Link>]{' '}
```
Comment (acd124) [inline src/components/projects/wbs-details/work-package-container/changes-list/changes-list.tsx:25]: Why was this changed to link to the wbs element that was changed? I believe this will always route back to itself, thus doing nothing, I think the intention was to link to the change request that resulted in the change to this wbs element, since the changes is a list of changes made to this work package.

File: src/components/projects/wbs-details/work-package-container/changes-list/changes-list.tsx [hunk 2]
```diff
@@ -20,8 +22,17 @@ const ChangesList: React.FC<ChangesListProps> = ({ changes }) => {
...
+          <OverlayTrigger
+            placement="right"
+            overlay={
+              <Tooltip id="tooltip">
+                {fullNamePipe(ic.implementer)} - {datePipe(new Date(ic.dateImplemented))}
```
Comment (acd124) [inline src/components/projects/wbs-details/work-package-container/changes-list/changes-list.tsx:30]: Do you need to make a new Date? I believe ImplementedChange.dateImplemented is already supposed be a Date

**Review by acd124 — CHANGES_REQUESTED** [review]
The overlay looks fine, I'd recommend looking at the other place where there's hover in the app for how you might write tests for it (`src/components/change-requests/change-request-details/change-request-details/review-notes/review-notes.test.tsx`)

File: src/components/projects/wbs-details/work-package-container/changes-list/changes-list.tsx (hunk 2 above)
Comment (RChandler234) [reply src/components/projects/wbs-details/work-package-container/changes-list/changes-list.tsx:30]: yeah, I added this while troubleshooting because datePipe wasn't recognizing it as a date

File: src/components/projects/wbs-details/work-package-container/changes-list/changes-list.tsx (hunk 1 above)
Comment (RChandler234) [reply src/components/projects/wbs-details/work-package-container/changes-list/changes-list.tsx:25]: yeah no clue, I'll switch it back

File: src/components/projects/wbs-details/work-package-container/changes-list/changes-list.tsx (hunk 2 above)
Comment (jamescd18) [reply src/components/projects/wbs-details/work-package-container/changes-list/changes-list.tsx:31]: why wasn't it? can you screenshot the error?

**Review by jamescd18 — CHANGES_REQUESTED** [review]

File: src/components/projects/wbs-details/work-package-container/changes-list/changes-list.tsx (hunk 2 above)
Comment (RChandler234) [reply src/components/projects/wbs-details/work-package-container/changes-list/changes-list.tsx]: image

Comment (acd124) [reply src/components/projects/wbs-details/work-package-container/changes-list/changes-list.tsx]: The transformers were never updated with ImplementedChange.dateImplemented, so it's just a string rn

Comment (RChandler234) [reply src/components/projects/wbs-details/work-package-container/changes-list/changes-list.tsx]: oh I see, I had just assumed the backend was working correctly and didn't check that, I'll go fix it

Comment (RChandler234) [reply src/components/projects/wbs-details/work-package-container/changes-list/changes-list.tsx]: thanks alexander

**Review by jamescd18 — APPROVED** [review]
Dope! LGTM!
